### ISSUE TITLE
BinaryCrossentropy and binary_crossentropy in the. same `tf.keras.losses` module

### ISSUE THREAD

**Issue by ben-xD** [tensorflow/tensorflow#43827]
## URL(s) with the issue:
https://www.tensorflow.org/api_docs/python/tf/keras/losses/binary_crossentropy
https://www.tensorflow.org/api_docs/python/tf/keras/losses/BinaryCrossentropy

## Description of issue (what needs changing):
There are 2 versions of Binary Cross Entropy, it would be less confusing to have just one.

Also, only `tf.keras.losses.binary_crossentropy` (or alternatively `"binary_crossentropy"`) works in the below code:
```python
model.compile(optimizer = RMSprop(lr=0.0001), 
              loss = tf.keras.losses.binary_crossentropy, 
              metrics = ["accuracy"])
```
I wonder why `"BinaryCrossentropy"` or `tf.keras.losses.BinaryCrossentropy` don't work.

**Comment by ranzer** [comment]
Which TF version are you using?
I tried following example on TF 2.3.0 and it works as expected:
```python
import numpy as np
import tensorflow as tf

print(tf.__version__)

y_true = [[0, 1], [0, 0]]
y_pred = [[0.6, 0.4], [0.4, 0.6]]

bce1 = tf.keras.losses.binary_crossentropy(y_true, y_pred)
bce2 = tf.keras.losses.BinaryCrossentropy(reduction=tf.keras.losses.Reduction.NONE)

assert np.all([ bce1.numpy(), bce2(y_true, y_pred).numpy() ]) == True
```

**Comment by ben-xD** [comment]
Hi ranzer

I believe I was confused by the difference between them (class vs function). Yes, if you instantiate `BinaryCrossentropy` first, then pass the data, it works.

So actually, `model.compile(optimizer="adam", metrics=['accuracy'], loss=tf.keras.losses.SparseCategoricalCrossentropy())` works, notice the extra `()` needed when using **BinaryCrossentropy**.
